### DISCUSSION TITLE
Different metric result in callback and checkpoint?

### DISCUSSION THREAD

**Discussion by rudaoshi** [Lightning-AI/pytorch-lightning#6352] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I print a log in on_validation_end in a callback:

```python
def on_validation_end(self, trainer, pl_module):

        if pl_module.global_rank == 0:
            logging.info("Dev: " + str(pl_module.problem_evaluator_dev.compute()))
```

and it print a log:

> INFO:root:Dev: {'unlabel.Precision': tensor(0.7071, device='cuda:0'), 'unlabel.Recall': tensor(0.7551, device='cuda:0'), 'unlabel.FBeta': tensor(0.7303, device='cuda:0'), 'label.Precision': tensor(0.7323, device='cuda:0'), 'label.Recall': tensor(0.7323, device='cuda:0'), 'label.FBeta': tensor(0.7323, device='cuda:0'), 'unlabel_graph_accuracy': tensor(0.3521, device='cuda:0'), 'label_graph_accuracy': tensor(0.3287, device='cuda:0')}

Following the log, the checkpoint callback print a log:

> Epoch 1, global step 1567: dev_label.FBeta/dataloader_idx_0 reached 0.81325 (best 0.81325), saving model to ".../lightning_logs/version_21/checkpoints/epoch=1-step=1567.ckpt" as top 1

where the "dataloader_idx_0" is corresponding to the  problem_evaluator_dev, so the printed value should be same. 

Can any one help to explain?

pytorch_lightning version 1.1.8.
